`jit`ing the `DnCNN` prox on GPU

When  I run `pytest scico/test/test_functional.py -k DnCNN` with a GPU install of SCICO, I get a failure in `TestDnCNN.test_prox`, which is comparing the `jit`ed to un`jit`ed version of the DnCNN prox. These results are disturbingly different:

    E       Mismatched elements: 1020 / 1024 (99.6%)
    E       Max absolute difference: 0.70825666
    E       Max relative difference: 83.99731
    E        x: array([[-1.311581e+00, -4.477247e-01, -2.993065e-01, ...,  1.231516e-01,
    E                2.924881e-01,  1.054722e-01],
    E              [ 2.548240e-01,  7.504759e-01, -3.427111e-01, ...,  1.717497e-01,...
    E        y: array([[-1.290338, -0.550369, -0.359681, ...,  0.004449,  0.33103 ,
    E                0.088125],
    E              [ 0.211299,  0.899408, -0.37546 , ...,  0.460677, -0.817596,...

Might this be a JAX bug? One could try bumping the jax version and see if it is fixed.

--- a/scico/test/test_functional.py
+++ b/scico/test/test_functional.py
@@ -477,6 +477,10 @@ class TestBM3D:
 
 
 class TestDnCNN:
+    import os
+
+    os.environ["XLA_FLAGS"] = "--xla_gpu_deterministic_reductions --xla_gpu_autotune_level=0"
+
     def setup(self):
         key = None
         N = 32

